src/snowflake/cli/plugins/nativeapp/codegen/snowpark/models.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import List, Optional
+
+from pydantic import Field
+from snowflake.cli.api.project.schemas.snowpark.callable import _CallableBase
+from snowflake.cli.api.project.schemas.updatable_model import IdentifierField
+
+
+class ExtensionFunctionTypeEnum(str, Enum):
+    PROCEDURE = "PROCEDURE"
+    FUNCTION = "FUNCTION"
+    TABLE_FUNCTION = "TABLE_FUNCTION"
+    AGGREGATE_FUNCTION = "AGGREGATE_FUNCTION"
+
+
+class NativeAppExtensionFunction(_CallableBase):
+    function_type: ExtensionFunctionTypeEnum
```

Comment:
Nit: just `type` is clearer as some of these aren't functions, they're procedures

Reply:
type isn't allowed in source code by the linter (it shadows a python built-in), but I made the change in the schema at least.

Reply:
Ah, I didn't realize that. OK with whatever in that case.